### PR TITLE
fio: clarify fio.basename() with slash at end

### LINKED ISSUE #2124: [2pt] fio.basename behavior doesn't conform to POSIX
**Product:** Tarantool
**Audience/target:** dev
**Root document:** https://www.tarantool.io/en/doc/latest/reference/reference_lua/fio

# Details

Tarantool version - 2.8.0-75-g6610bce9b (but actually all)

```console
tarantool> fio.dirname('/x/')
---
- /
...

tarantool> fio.basename('/x/')
---
- 
...
```

Expected behavior

```
$ dirname '/x/'
/
$ basename '/x/'
x
```

See also https://pubs.opengroup.org/onlinepubs/9699919799/

### PR DESCRIPTION
**Pull request by Totktonada**
Fixes #2124

### CODE REVIEW

**Comment by Totktonada** [comment]
(I'm not sure that the wording is nice, but it is my best tonight. I would like to have this caution in the documentation and open for proposals.)

**Comment by Totktonada** [comment]
https://docs.d.tarantool.io/en/doc/gh-2124-fio-basename-trailing-slash/reference/reference_lua/fio/#fio-basename

image

File: doc/reference/reference_lua/fio.rst [hunk 1]
```diff
@@ -197,6 +197,10 @@ Below is a list of all ``fio`` functions and members.
     Given a full path name, remove all but the final part (the file name).
     Also remove the suffix, if it is passed.
 
+    Note that the basename of a path with the trailing slash is the empty
```
Comment (patiencedaur) [inline doc/reference/reference_lua/fio.rst]:
```suggestion
    Note that the basename of a path with a trailing slash is an empty
```

Comment (Totktonada) [reply doc/reference/reference_lua/fio.rst]: I guess I should also replace

> Example with the trailing slash:

with

> Example with a trailing slash:

Comment (Totktonada) [reply doc/reference/reference_lua/fio.rst]: Updated.

**Pull request merged**